docs/schema.md

```diff
@@ -289,6 +289,117 @@ you must also specify a `type`, described in
 Use the following types to indicate how the properties in your schema are
 treated by GCP Marketplace.
 
+### `schemaVersion`
```

Comment:
Note that the entire section here is only relevant for schema v2 (i.e. `applicationApiVersion`, `publishedVersion`, `images`).

So I think we should have a section earlier in the file making the user choose between v1 and v2. We would recommend v2, saying that it will be required starting December 2019. And have 2 sections about v1 and v2. The properties, other than the `x-google-marketplace.type=IMAGE` ones are shared between v1 and v2.

The changes are also summarized [here](https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blob/master/docs/managed-updates.md#update-your-schema). You can also compare an example of [v2](https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blob/master/tests/marketplace/deployer_envsubst_base/standard_v2/schema.yaml) and [v1](https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blob/master/tests/marketplace/deployer_envsubst_base/standard/schema.yaml).

At a high level:
- The fields you added here are all v2 only.
- Only v1 requires `x-google-marketplace.type=IMAGE` property.
- v1 also has `applicationApiVersion` at the top level, which has been moved to under the top level `x-google-marketplace` in v2.

Reply:
I think I've addressed this here - let me know if this works for you or not!